pkg/nuclide-swift/lib/taskrunner/SwiftPMTaskRunner.js

```diff
@@ -0,0 +1,185 @@
+  id: string;
+  name: string;
+  _disposables: CompositeDisposable;
+  _store: SwiftPMTaskRunnerStore;
+  _actions: SwiftPMTaskRunnerActions;
+  _taskList: Observable<Array<TaskMetadata>>;
+  _outputMessages: Subject<Message>;
+
+  constructor(initialState: ?SwiftPMTaskRunnerStoreState) {
+    this.id = 'swiftpm';
+    this.name = 'Swift';
+
+    const dispatcher = new Dispatcher();
+    this._store = new SwiftPMTaskRunnerStore(dispatcher, initialState);
+    this._actions = new SwiftPMTaskRunnerActions(dispatcher);
+    this._outputMessages = new Subject();
+
+    this._disposables = new CompositeDisposable();
+    this._disposables.add(this._store);
+    this._disposables.add(this._outputMessages);
+  }
+
+  dispose(): void {
+    this._disposables.dispose();
+  }
+
+  serialize(): SwiftPMTaskRunnerStoreState {
+    return this._store.serialize();
+  }
+
+  getExtraUi(): ReactClass<any> {
+    const store = this._store;
+    const actions = this._actions;
+    return class ExtraUi extends React.Component {
+      props: {
+        activeTaskType: ?string,
+      };
+
+      render(): React.Element<any> {
+        return (
+          <SwiftPMTaskRunnerToolbar
+            store={store}
+            actions={actions}
+            activeTaskType={this.props.activeTaskType}
+          />
+        );
+      }
+    };
+  }
+
+  observeTaskList(callback: (taskList: Array<TaskMetadata>) => mixed): IDisposable {
+    if (this._taskList == null) {
+      this._taskList = Observable.of(SwiftPMTaskRunnerTaskMetadata);
+    }
+    return new DisposableSubscription(
+      this._taskList.subscribe({next: callback}),
+    );
+  }
+
+  getIcon(): ReactClass<any> {
+    return SwiftIcon;
+  }
+
+  runTask(taskName: string): Task {
+    const chdir = this._store.getChdir();
+    const configuration = this._store.getConfiguration();
+    const buildPath = this._store.getBuildPath();
+
+    let command;
+    switch (taskName) {
+      case SwiftPMTaskRunnerBuildTaskMetadata.type:
+        command = buildCommand(
+          chdir,
+          configuration,
+          this._store.getXcc(),
+          this._store.getXlinker(),
+          this._store.getXswiftc(),
+          buildPath,
+        );
+        break;
+      case SwiftPMTaskRunnerTestTaskMetadata.type:
+        command = testCommand(chdir, buildPath);
+        break;
+      default:
+        throw new Error(`Unknown task name: ${taskName}`);
+    }
+
+    atom.commands.dispatch(atom.views.getView(atom.workspace), 'nuclide-console:show');
+    this._logOutput(`${command.command} ${command.args.join(' ')}`, 'log');
+
+    const observable = observeProcess(
+      () => safeSpawn(command.command, command.args),
+    ).do(message => {
+      switch (message.kind) {
+        case 'stderr':
+        case 'stdout':
+          this._logOutput(message.data, 'log');
+          break;
+        case 'exit':
+          this._logOutput(`Exited with exit code ${message.exitCode}`, 'log');
```

Comment:
Wow, that looks awesome. Thanks!

![screen shot 2016-07-20 at 9 38 14 pm](https://cloud.githubusercontent.com/assets/552921/17008757/544188a2-4ec2-11e6-9721-d3b4a0f8c016.png)